api/api/middleware/client_application_middleware.py

```diff
@@ -0,0 +1,16 @@
+from api.utils.oauth2_helper import get_token_info
+
+
+def client_application_middleware(get_response):
```

Comment:
A few docstrings about this middleware would be nice! Just what's in the PR description is good enough.

Reply:
Good idea. I'll rename this to add "headers" in there as well. The name I've chosen to commit is bafflingly obscure :sweat_smile: